frontend/app/src/connection/StaticConnection.tsx

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2025)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  ForwardMsgList,
+  localStorageAvailable,
+  logError,
+} from "@streamlit/lib"
+
+import { ConnectionState } from "./ConnectionState"
+
+// Holds url for static asset location
+export const STATIC_ASSET_CONFIG = "https://data.streamlit.io/static.json"
+
+type OnMessage = (ForwardMsg: any) => void
+type OnConnectionStateChange = (
+  connectionState: ConnectionState,
+  errMsg?: string
+) => void
+
+interface Props {
+  /** The static notebook's ID from query param */
+  staticNotebookId: string
+
+  /**
+   * Function called when our ConnectionState changes.
+   * For a StaticConnection, used for StatusWidget
+   */
+  onConnectionStateChange: OnConnectionStateChange
+
+  /**
+   * Function called when we receive a new ForwardMsg
+   */
+  onMessage: OnMessage
+}
+
+// Fetches the static asset url from the config file
+export async function getStaticConfig(): Promise<string> {
+  const isLocalStoreAvailable = localStorageAvailable()
+  let staticAssetUrl = ""
+
+  // Pull static asset url from localStorage if available
+  if (isLocalStoreAvailable) {
+    const cachedConfig = window.localStorage.getItem("stStaticAssetUrl")
+    if (cachedConfig) {
+      staticAssetUrl = cachedConfig
```

Comment:
I would just early return here.

`return cachedConfig`

And probably call it `cachedStaticAssetUrl`